src/Components/WebAssembly/WebAssembly/src/HotReload/HotReloadAgent.cs

```diff
@@ -236,44 +140,47 @@ private static Type[] GetMetadataUpdateTypes(IReadOnlyList<UpdateDelta> deltas)
                 continue;
             }
 
-            var assemblyTypes = assembly.GetTypes();
-
-            foreach (var updatedType in delta.UpdatedTypes ?? Array.Empty<int>())
+            foreach (var updatedType in delta.UpdatedTypes)
             {
-                var type = assemblyTypes.FirstOrDefault(t => t.MetadataToken == updatedType);
-                if (type != null)
+                // Must be a TypeDef.
+                Debug.Assert(updatedType >> 24 == 0x02);
+
```

Comment:
Nit: it might be nice to include a comment explaining where these numbers come from (I'm assuming [this](https://github.com/dotnet/runtime/blob/234e2f7ec5dd315014f31574719900c0715f9477/src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/TableIndex.cs#L10) and [this](https://github.com/dotnet/runtime/blob/234e2f7ec5dd315014f31574719900c0715f9477/src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Internal/MetadataFlags.cs#L331))?